jscomp/ml/record_spread.ml

```diff
@@ -0,0 +1,51 @@
+module StringMap = Map.Make (String)
+
+let t_equals t1 t2  = t1.Types.level = t2.Types.level && t1.id = t2.id
+
+let substitute_types ~type_map (t : Types.type_expr) =
+  if StringMap.is_empty type_map then t
+  else
+    let apply_substitution type_variable_name t =
+      match StringMap.find_opt type_variable_name type_map with 
+      | None -> t
+      | Some substituted_type -> substituted_type
+    in
+    let rec loop (t : Types.type_expr) =
+      match t.desc with
+      | Tlink t -> loop t
```

Comment:
might be safer to respect the original type here and change nothing: `Tlink (loop t)`